utils/doFetch.mjs

```
@@ -0,0 +1,24 @@
+import { getAuthToken } from "./handleToken.mjs";
+
+export async function doFetch(url, isAuth = false, options = {}){
+    try{
+        const headers = {
+            'Content-Type': 'application/json',
+        };
+        if(isAuth){
+            const authToken = getAuthToken();
+            headers["Authorization"] =`Bearer ${authToken}`;
+        };
+
+        const combinedOptions ={headers, ...options};
+        console.log(combinedOptions);
+        const response = await fetch(url, combinedOptions);
+        const data = await response.json();
+        return data;
+    }catch(error){
+        console.log(error);
+        throw error
+    }finally{
+        //
+    }
```

1) Lines of code that are not being used should be removed - referring to `lines 21 - 23`.

2) `catch block`: to suppress an error
     `throw`: to create an error

      Understand the differences, they have different purposes. Here you're basically catching an error 
      and throwing it again.

      **Suggestion:** remove the throw from the catch block and use the try/catch block in a UI/UX code 
      block e.g event listener.